test/powershell/Modules/Microsoft.PowerShell.Utility/WebCmdlets.Tests.ps1

```diff
@@ -2188,3 +2189,10 @@ Describe "Web cmdlets tests using the cmdlet's aliases" -Tags "CI" {
         $result.Hello | Should Be "world"
     }
 }
+
+Describe "PSUserAgent Tests" {
+    It "App Should Match ^PowerShell/\d+\.\d+\.\d+.*" {
+        $app = [Microsoft.PowerShell.Commands.PSUserAgent].GetProperty('App',@('Static','NonPublic')).GetValue($null,$null) 
+        $app | Should Match '^PowerShell/\d+\.\d+\.\d+.*'
```

Comment:
Here I don't see the point of checking internal property - we are free to change (add/remove) internals in any time. We should test public API. Here "public API" is User-Agent which we send to a server and get it back. We should test the User-Agent format if we want to control the User-Agent. So it is good to use regex pattern. I think that check User-Agent 14 times it's redundant - we need to have one check for each web cmdlet.

Reply:
Again, the point of most of those tests is not to test the User-Agent that is sent. Most of those assertions are being used to ensure meaningful data is returned in a response after a set of parameters has been used to make web requests. The only reason I updated them at all was because the change from `WindowsPowerShell` to `PowerShell` would have caused all these tests to false fail.

I will change the two tests that are explicitly about the User-Agent.

Reply:
As you mentioned `$uri.Authority` is sufficient. So I'd remove `$DefaultUserAgent` at all. I unlike it as you see 😄

Reply:
I'm ok with that. This entire file needs to be more deeply analyzed and cleaned up. Maybe after I'm done making radical changes to it every day 😆.